packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ProtectedListView/index.js

```diff
@@ -1,14 +1,23 @@
 import React from 'react';
 
 import { CheckPagePermissions } from '@strapi/helper-plugin';
+import { useSelector } from 'react-redux';
 
-import adminPermissions from '../../../../../permissions';
+import { selectAdminPermissions } from '../../../../App/selectors';
 import ListView from '../ListView';
 
-const ProtectedApiTokenListView = () => (
-  <CheckPagePermissions permissions={adminPermissions.settings['api-tokens'].main}>
-    <ListView />
-  </CheckPagePermissions>
-);
+const ProtectedApiTokenListView = () => {
+  const permissions = useSelector(selectAdminPermissions);
+
+  if (!permissions?.settings?.['api-tokens']?.main) {
+    return null;
+  }
```

Comment:
this seems odd? we wouldn't render anything and the user would have no feedback – doesn't the `CheckPagePermissions` component handle this?

Reply:
Good catch. It is leftover and shouldn't be there.

Reply:
Removed & rebased.